app/models/manageiq/providers/ansible_tower/automation_manager/cloud_credential.rb

```diff
@@ -1,2 +1,10 @@
 class ManageIQ::Providers::AnsibleTower::AutomationManager::CloudCredential < ManageIQ::Providers::AutomationManager::Authentication
+  def self.credential_types
+    credential_types = {}
+    subclasses.each do |subclass|
+      provider_name = subclass.name.demodulize.split('Credential').first
+      credential_types[provider_name] = subclass.name
+    end
+    credential_types.sort.to_h
```

Comment:
If sorting is important, it would be more efficient to sort before creating the hash.
```ruby
subclasses.sort_by(&:name).each_with_object.....
```